packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -36,23 +48,23 @@ export default class OperatorModeStateService extends Service {
   addItemToStack(item: StackItem) {
     let stackIndex = item.stackIndex;
     if (!this.state.stacks[stackIndex]) {
-      this.state.stacks[stackIndex] = new TrackedObject({
-        items: new TrackedArray([]),
-      });
+      this.state.stacks[stackIndex] = new TrackedArray([]);
+    }
+    this.state.stacks[stackIndex].push(item);
+    if (item.type === 'card') {
+      this.addRecentCards(item.card);
     }
-    this.state.stacks[stackIndex].items.push(item);
-    this.addRecentCards(item.card);
     this.schedulePersist();
   }
 
   removeItemFromStack(item: StackItem) {
     let stackIndex = item.stackIndex;
-    let itemIndex = this.state.stacks[stackIndex].items.indexOf(item);
-    this.state.stacks[stackIndex].items.splice(itemIndex);
+    let itemIndex = this.state.stacks[stackIndex].indexOf(item);
+    this.state.stacks[stackIndex].splice(itemIndex); // Always remove anything above the item
```

Comment:
👍 
 
 Thanks. I actually think we can rename this
 
 `removeItemFromStack(item: StackItem)` -> `popItemsFromStack(lowestItem: StackItem)`

Reply:
I wonder if actually we should not allow removing an item at any index. Maybe only pop should be permitted, as removing an item at any index could destabilize contains pointers.

Reply:
> This doesn’t pop though. It can remove an item at any index. We already do have a pop though that we added.

`splice(index,1)` removes a particular element
`splice(index)` removes index and all elements at the end. Hence, why I suggested the word "pop" (like popMany kinda thing)

Reply:
> I wonder if actually we should not allow removing an item at any index. Maybe only pop should be permitted, as removing an item at any index could destabilize contains pointers.

I do agree with this.

Reply:
Ah thanks for reminding me! That’s the 2nd time I missed that. That `.splice(index,1)` is very problematic. We should probably not be doing that. `pop()` generally means just remove the last item. (This is how `Array.pop()` works).

Reply:
i did a quick review--the `splice(index, 1)` is actually _not_ a problem. the splice is not actually happening on the stack. rather the splice is happening in the array that holds all the stacks. basically we splice out the stack from the array of stacks. so that logic is totally cool. but yeah a rename of this method would be nice. since we already have a `pop` which removes the last stack item, I'll call this `trim` since it removes trailing stack items